operator/controllers/btpoperator_controller.go

```diff
@@ -143,17 +232,37 @@ func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v
 	ready, err := manifest.InstallChart(&logger, installInfo, []types.ObjectTransform{r.labelTransform})
 	if err != nil {
 		logger.Error(err, fmt.Sprintf("error while installing resource %s", client.ObjectKeyFromObject(cr)))
-		cr.SetStatus(status.WithState(types.StateError))
-		return r.Status().Update(ctx, cr)
+		return r.SetStatus(types.StateError, ctx, cr)
 	}
 	if ready {
-		cr.SetStatus(status.WithState(types.StateReady))
-		return r.Status().Update(ctx, cr)
+		return r.SetStatus(types.StateReady, ctx, cr)
 	}
 
 	return nil
 }
 
+func (r *BtpOperatorReconciler) HandleErrorState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Handling Error state")
+
+	return r.SetStatus(types.StateProcessing, ctx, cr)
+}
+
+func (r *BtpOperatorReconciler) HandleDeletingState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+	logger := log.FromContext(ctx)
+	if err := r.handleDeprovisioning(ctx); err != nil {
+		logger.Error(err, "deprovisioning failed")
+		return r.SetStatus(types.StateError, ctx, cr)
+	} else {
+		logger.Info("deprovisioning success. clearing finalizers for btp manager")
+		cr.SetFinalizers([]string{})
+		if err := r.Update(ctx, cr); err != nil {
+			return err
```

Comment:
maybe it's worth to provide more context to the error thrown from Update() using `fmt.Errorf()`